### PR TITLE
Refactor project structure and ORM enums

### PR DESCRIPTION
**Pull request by omgronny**

### CODE REVIEW

**Comment by IP-13** [comment]
|Overall Project|33.06% **`-23.33%`**|:x:|
|:-|:-|:-:|
|Files changed|0%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[TaskController.java](https://github.com/itmo-events-app/event-app/blob/9bd60af56978f4d73b77ba3b86571f6bcd2dc07b/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FTaskController.java)|3.85% **`-28.21%`**|:x:|
|[EventService.java](https://github.com/itmo-events-app/event-app/blob/9bd60af56978f4d73b77ba3b86571f6bcd2dc07b/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventService.java)|3.16% **`-65.26%`**|:x:|
|[EventRequest.java](https://github.com/itmo-events-app/event-app/blob/9ce0536b899b66d0cd3f79896a9e2f66dc016bdd/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Frequest%2FEventRequest.java)|0%|:green_apple:|

**Comment by IP-13** [comment]
|Overall Project|33.06% **`-23.33%`**|:x:|
|:-|:-|:-:|
|Files changed|0%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[TaskController.java](https://github.com/itmo-events-app/event-app/blob/5b0defcc2c8ba78901c6273d216c66418f5b1076/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FTaskController.java)|3.85% **`-28.21%`**|:x:|
|[EventService.java](https://github.com/itmo-events-app/event-app/blob/5b0defcc2c8ba78901c6273d216c66418f5b1076/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventService.java)|3.16% **`-65.26%`**|:x:|
|[EventRequest.java](https://github.com/itmo-events-app/event-app/blob/9ce0536b899b66d0cd3f79896a9e2f66dc016bdd/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Frequest%2FEventRequest.java)|0%|:green_apple:|

**Comment by IP-13** [comment]
|Overall Project|38.51% **`-27.18%`**|:x:|
|:-|:-|:-:|
|Files changed|0%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[EventController.java](https://github.com/itmo-events-app/event-app/blob/205bfcfead1c8707239196dfffb3b6eb09c5484e/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FEventController.java)|37.5%|:green_apple:|
|[TaskController.java](https://github.com/itmo-events-app/event-app/blob/205bfcfead1c8707239196dfffb3b6eb09c5484e/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FTaskController.java)|3.85% **`-28.21%`**|:x:|
|[EventService.java](https://github.com/itmo-events-app/event-app/blob/205bfcfead1c8707239196dfffb3b6eb09c5484e/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventService.java)|3.16% **`-65.26%`**|:x:|

File: build.gradle.kts [hunk 1]
```diff
@@ -29,6 +29,7 @@ dependencies {
     implementation("org.springframework.boot:spring-boot-starter-web")
     implementation("org.flywaydb:flyway-core:9.22.3")
     implementation("org.springframework.boot:spring-boot-starter-validation")
+    implementation("io.hypersistence:hypersistence-utils-hibernate-63:3.7.3")
```
Comment (IP-13) [inline build.gradle.kts:32]: У меня без этой штуки находит тоже, так что лучше удалить

File: src/main/java/org/itmo/eventapp/main/model/dto/EventRequest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import jakarta.persistence.*;
+import lombok.Data;
+import org.itmo.eventapp.main.model.entity.Event;
+import org.itmo.eventapp.main.model.entity.EventFormat;
+import org.itmo.eventapp.main.model.entity.EventStatus;
+import org.itmo.eventapp.main.model.entity.Place;
+
+import java.time.LocalDateTime;
+
+// TODO: Add correct annotations
```
Comment (IP-13) [inline src/main/java/org/itmo/eventapp/main/model/dto/EventRequest.java:12]: Речь про аннотации валидации или о чем?

File: src/main/java/org/itmo/eventapp/main/repo/TaskRepo.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package org.itmo.eventapp.main.repo;
+
+import org.itmo.eventapp.main.model.entity.Task;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface TaskRepo extends JpaRepository<Task,Integer>{
```
Comment (IP-13) [inline src/main/java/org/itmo/eventapp/main/repo/TaskRepo.java:6]: Нужна еще аннотация @Repository

Comment (IP-13) [reply src/main/java/org/itmo/eventapp/main/repo/TaskRepo.java:6]: Давай лучше называть не repo, а repository, и пакет тоже repository

Comment (IP-13) [reply src/main/java/org/itmo/eventapp/main/repo/TaskRepo.java:6]: Ваще удали его, я ща ПР сделаю у меня уже это есть

File: src/main/java/org/itmo/eventapp/main/model/dto/EventRequest.java (hunk 2 above)
Comment (omgronny) [reply src/main/java/org/itmo/eventapp/main/model/dto/EventRequest.java:12]: ага

**Comment by IP-13** [comment]
|Overall Project|38.51% **`-27.18%`**|:x:|
|:-|:-|:-:|
|Files changed|0%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[EventController.java](https://github.com/itmo-events-app/event-app/blob/4776c09170fe4cb5d9f76ecae16acdf3fabec653/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FEventController.java)|37.5%|:green_apple:|
|[TaskController.java](https://github.com/itmo-events-app/event-app/blob/4776c09170fe4cb5d9f76ecae16acdf3fabec653/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FTaskController.java)|3.85% **`-28.21%`**|:x:|
|[EventService.java](https://github.com/itmo-events-app/event-app/blob/4776c09170fe4cb5d9f76ecae16acdf3fabec653/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventService.java)|3.16% **`-65.26%`**|:x:|

**Pull request merged**